### PR TITLE
Refactor DomainId change user interaction [15082] 

### PR DESCRIPTION
**Pull request by jparisu**

### CODE REVIEW

File: plugins/datastreamer_plugin/ui/domain_selection_dialog/dialogselectdomain.cpp [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+namespace eprosima {
+namespace plotjuggler {
+namespace ui {
+
+DialogSelectDomain::DialogSelectDomain(
+        const uint32_t current_domain_id,
+        QWidget* parent /* = nullptr */)
+    : QDialog(parent)
+    , ui(new Ui::DialogSelectDomain)
```
Comment (antonrv) [inline plugins/datastreamer_plugin/ui/domain_selection_dialog/dialogselectdomain.cpp]: Seems we are leaking memory here (unless it is internally handled within QT somehow, add a comment in that case), as `ui` is not a `unique_ptr` and I don't see any `delete`. Easier way is to wrap it around `unique_ptr`

File: plugins/datastreamer_plugin/ui/topic_selection_dialog/dialogselecttopics.cpp [hunk 2]
```diff
@@ -177,13 +178,30 @@ void DialogSelectTopics::on_change_domain_button_clicked()
 {
     DEBUG("Calling on_change_domain_button_clicked");
 
-    // Change domain connected
-    domain_id_selected_ =
-            static_cast<unsigned int>(ui->domainid_spin->value());
+    // Start a new dialog to select a domain
+    DialogSelectDomain* change_domain_dialog = new DialogSelectDomain(domain_id_connected_, this);
```
Comment (antonrv) [inline plugins/datastreamer_plugin/ui/topic_selection_dialog/dialogselecttopics.cpp:182]: please use safer RAII semantics for safety
either using `unique_ptr`:
`auto change_domain_dialog = std::make_unique<DialogSelectDomain>(domain_id_connected_, this);`

or plain object if DialogSelectDomain is not polymorphic:
`DialogSelectDomain change_domain_dialog(domain_id_connected_, this);`

File: plugins/datastreamer_plugin/ui/topic_selection_dialog/dialogselecttopics.cpp [hunk 3]
```diff
@@ -177,13 +178,30 @@ void DialogSelectTopics::on_change_domain_button_clicked()
...
+        {
+            // Call listener to connect to domain
+            listener_->on_domain_connection(
+                static_cast<unsigned int>(domain_id));
+        }
+    }
+    // In case the dialog is cancelled, do nothing
+
+    // Delete the dialog
+    delete change_domain_dialog;
```
Comment (antonrv) [inline plugins/datastreamer_plugin/ui/topic_selection_dialog/dialogselecttopics.cpp:204]: unneeded `delete` with RAII

File: plugins/datastreamer_plugin/ui/domain_selection_dialog/dialogselectdomain.h [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+    void on_buttonBox_rejected();
+
+protected:
+
+    void check_accept_enable_();
+
+    const uint32_t current_domain_id_;
+    uint32_t selected_domain_id_;
+
+    Ui::DialogSelectDomain* ui;
```
Comment (antonrv) [inline plugins/datastreamer_plugin/ui/domain_selection_dialog/dialogselectdomain.h:67]: Much better to use `std::unique_ptr` if `ui` is polymorphic or just a data member if not

**Review by antonrv — CHANGES_REQUESTED** [review]
Minor RAII-related issues

File: plugins/datastreamer_plugin/ui/domain_selection_dialog/dialogselectdomain.cpp (hunk 1 above)
Comment (jparisu) [reply plugins/datastreamer_plugin/ui/domain_selection_dialog/dialogselectdomain.cpp]: _Qt works in mysterious ways._
https://stackoverflow.com/a/34434321/17049427

File: plugins/datastreamer_plugin/ui/topic_selection_dialog/dialogselecttopics.cpp (hunk 2 above)
Comment (jparisu) [reply plugins/datastreamer_plugin/ui/topic_selection_dialog/dialogselecttopics.cpp:182]: This follows Qt guides. I rather leave it this way.

File: plugins/datastreamer_plugin/ui/topic_selection_dialog/dialogselecttopics.cpp (hunk 3 above)
Comment (jparisu) [reply plugins/datastreamer_plugin/ui/topic_selection_dialog/dialogselecttopics.cpp:204]: answered before

File: plugins/datastreamer_plugin/ui/domain_selection_dialog/dialogselectdomain.h (hunk 4 above)
Comment (jparisu) [reply plugins/datastreamer_plugin/ui/domain_selection_dialog/dialogselectdomain.h:67]: answered before

**Comment by jparisu** [comment]
New ui change added to change style of the dds configuration. Please review it.

**Review by antonrv — APPROVED** [review]

**Pull request merged**